### ISSUE TITLE
Keyboard shortcut: switch mode when editing label layers

### ISSUE THREAD

**Issue by hungyiwu** [napari/napari#1345]
## To Reproduce

1. Open napari viewer, create a new label layer
2. Press `f` goes to fill mode, and press space goes to pan/zoom mode, looking good
3. Press `f` to go to fill mode and fill one label area
4. Now press space, the mode status switched to pan/zoom mode for one sec, and them jumped back to fill mode. Reproducible at this stage
5. Mouse click at the magnifier icon to go to pan/zoom mode and pan/zoom. Now pressing `f` goes to fill mode and pressing space again firmly changed back to pan/zoom mode

## Expected behavior

Mode switch through keyboard as stable as mouse clicking the icons

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

```
napari: 0.3.4
Platform: Linux-5.4.0-7634-generic-x86_64-with-debian-bullseye-sid
Python: 3.7.6 | packaged by conda-forge | (default, Jun 1 2020, 18:57:50) [GCC 7.5.0]
Qt: 5.12.5
PyQt5: 5.12.3
NumPy: 1.18.1
SciPy: 1.4.1
Dask: 2.17.2
VisPy: 0.6.4

GL version: 4.6 (Compatibility Profile) Mesa 20.0.4
MAX_TEXTURE_SIZE: 16384

Plugins:
- napari-plugin-engine: 0.1.5
- svg: 0.1.3
```

 - Any other relevant information:

Ubuntu 20.04
conda 4.8.3

**Comment by sofroniewn** [comment]
Unfortunately I think this is a duplicate of #133 and #1143 as you are on linux and we've had problems with "hold" nature of the spacebar - likely due to some problem here https://github.com/napari/napari/blob/d52ddf87d90328a8a2c1757d43e3075bb672732f/napari/_qt/qt_viewer.py#L555, but I'm not sure.

At a minimum though we should add something to our docs alerting people that this is a known bug until it is fixed. My appologies

**Comment by sofroniewn** [comment]
This should now be closed by #1350, do you want to try and see if it is fixed now?

**Comment by hungyiwu** [comment]
It worked! (the `master branch` installed using `python setup.py`) Thank you! I guess it'll be included in 0.3.5?

**Comment by sofroniewn** [comment]
Yay! Yes, it will be out in `0.3.5` which should come early next week.

Great work by @ziyangczi fixing this one, it has closed three issues!!
